.github/workflows/plonk-artifacts.yml

```diff
@@ -0,0 +1,92 @@
+# Generates and uploads new Plonk artifacts to S3, named `SPHINX_CIRCUIT_VERSION.tar.gz` unless manually overriden
+# On `push`, uploads new parameters only if there's no matching `SPHINX_CIRCUIT_VERSION.tar.gz` in S3
+# On `workflow_dispatch`, force-uploads new parameters regardless of existing files in S3
+name: Update Plonk artifacts in S3
+
+on:
+  push:
+    branches: dev
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Artifact version used for archive name, e.g. `v1.0.0-testnet`. Defaults to `SPHINX_CIRCUIT_VERSION` in `core/src/lib.rs`"
+        type: string
+        required: false
+        # Set to `SPHINX_CIRCUIT_VERSION` in subsequent step
+        default: ""
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  update-plonk-artifacts:
+    name: Update Plonk artifacts
+    runs-on: warp-ubuntu-latest-x64-32x
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: argumentcomputer/ci-workflows
+      - uses: ./.github/actions/ci-env
+      - uses: actions/checkout@v4
+      - name: Setup CI
+        uses: ./.github/actions/setup
+        with:
+          pull_token: ${{ secrets.REPO_TOKEN }}
+      - name: Install AWS CLI
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y awscli
+          echo "AWS_REGION=us-east-2" | tee -a $GITHUB_ENV
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          aws-access-key-id: ${{ secrets.S3_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.S3_SECRET_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Check for existing artifacts
+        id: check-s3
+        run: |
+          BUCKET_NAME="sphinx-plonk-params"
+          VERSION=$(grep -r "const SPHINX_CIRCUIT_VERSION" core/src/lib.rs | awk -F '"' '{print $2}')
```

Comment:
Could we add a line on a non-important lint job that runs on every PR (e.g. where we run `cargo fmt` or `cargo clippy`), that just checks the right-hand side of this assignment is non-empty? 

I'm worried about this file moving in a future PR and us forgetting to update this CI job.
A comment on the lint check would remind us to go update this plonk-artifacts job when that happens.